### DISCUSSION TITLE
Does `openapi-format` support sorting the path alphabetically?

### DISCUSSION THREAD

**Discussion by tuananhlai** [thim81/openapi-format#90] [Q&A] [answered]
Hi, thank you for creating such an amazing tool.

I have a question regarding the sorting feature. Does this tool support sorting the paths alphabetically? For example, if I have the following yaml:

```yaml
paths:
  /api/posts/{id}:
  /api/posts:
  /api/posts/{id}/comments:
```

Does this tool allow me to sort the paths and make it like so?

```yaml
paths:
  /api/posts:
  /api/posts/{id}:
  /api/posts/{id}/comments:
```

Thank you.

**Comment by thim81** [comment]
hi @tuananhlai 

At this moment, it does not sort the paths, since we consider this part of the developer experience (DX), where the API designer might put them in a specific order of important or common-sense.

We are extending the sorting of inline query params & headers, to facilitate "diffing" of generated specs.

Could you share a bit why you would want to sort the "paths"?

**Reply by tuananhlai** [reply]
Hi @thim81,

For OpenAPI files where the paths are not organized in any logical order, I think having an option to sort the paths alphabetically would significantly improve its readability.

**Reply by thim81** [reply]
hi @tuananhlai 

FYI, I'm working on extending openapi-format to allow the sorting of the paths by alphabet or by tag name.
I'm looking for a bit of input on how to "configure" it and would appreciate some input from users like you.

How would you envision configuring the order of paths by ...?

My idea was to add "sortPaths" property to be passed along via --sortFile.
Since openapi-format works with configs in files, and this new option influences the sorting, it would make sense to extend it like so.

**Reply by thim81** [reply, marked as answer]
hi @tuananhlai 

We just release 1.22.0 of openapi-format, which includes the option to sort paths alphabetically or by tag.

**Reply by tuananhlai** [reply]
Thank you very much 🙇
